src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -120,42 +125,50 @@ public FormValidation doCheckReturnStatus(@QueryParameter boolean returnStdout,
             return FormValidation.ok();
         }
 
+        @Override public final Set<? extends Class<?>> getRequiredContext() {
+            return ImmutableSet.of(FilePath.class, EnvVars.class, Launcher.class, TaskListener.class);
+        }
+
     }
 
     /**
      * Represents one task that is believed to still be running.
      */
-    @Restricted(NoExternalUse.class)
     @SuppressFBWarnings(value="SE_TRANSIENT_FIELD_NOT_RESTORED", justification="recurrencePeriod is set in onResume, not deserialization")
-    public static final class Execution extends AbstractStepExecutionImpl implements Runnable {
+    static final class Execution extends StepExecution implements Runnable {
```

Comment:
Unfortunately, this does seem to cause a `InvalidClassException`:

    Class does not extend stream superclass

Looking to see if there is some way around that which does not require going back to `AbstractStepExecutionImpl`.